### ISSUE TITLE
Need support for multi level drilldown menu

### ISSUE THREAD

**Issue by karthikjeeyar** [patternfly/patternfly#2635]
**Describe the issue. What is the expected and unexpected behavior?**
 Openshift Developer Console needs multi level dropdown menu in the topology component. 

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**

Enhancement

**What is your product and what release version are you targeting?**
 Openshift Developer Console ( Targetting for release version 4.4)

**Comment by mcarrano** [comment]
Rather than the fly-out approach, we want to implement this as a drill-down menu per designs from @maryshak1996 here: https://marvelapp.com/5cjajdg/screen/63256946 This behaves similar to a mobile tree menu where clicking into a parent node always exposes the child menu and the user clicks a back link to return to the parent.

![Screen Shot 2020-03-10 at 11 24 40 AM](https://user-images.githubusercontent.com/6619860/76328556-d467bd00-62c1-11ea-98e6-7a343ad844ff.png)

![Screen Shot 2020-03-10 at 11 25 07 AM](https://user-images.githubusercontent.com/6619860/76328615-e5b0c980-62c1-11ea-86ba-f89eb05f0233.png)

**Comment by ajacobs21e** [comment]
@maryshak1996 I have a drill-down menu use case that will have around 3-5 levels on average, with some having 10+. Is this how the menu would work with multiple upper levels 'stacking' at the top of the menu?

![drill down](https://user-images.githubusercontent.com/6415742/78074892-851d2580-7371-11ea-8066-be93e0a9d40a.png)

**Comment by maryshak1996** [comment]
Hey @ajacobs21e ! maybe I'm misunderstanding your use case here, but the content above the line there is really just set aside for a search (optional) and then JUST the parent of the items under the line. To access other 'siblings' to that parent element, the user would click on that one parent item, which would then bring them up one level. This component should drill in one level at a time. 
So I'm curious what the two 'parent' level items are in your screenshot above the line?

**Comment by ajacobs21e** [comment]
![drilldown](https://user-images.githubusercontent.com/6415742/78292043-6bf0b200-74f4-11ea-93ba-fcb4a94fb131.png)
@maryshak1996 here's the flow. The top row is how I was imagining it - for every level you go down, the parent levels stack. That allows the user to quickly go back to the beginning and also gives context to where you are.

The bottom row sounds more like how you intended it - only one parent level shows at a time. For my use case I would really like a way to go back to the beginning, so at the very least showing the previous parent + a "go to top" link. 
How would you recommend handling a use case with many levels? Thanks!

**Comment by mcarrano** [comment]
@ajacobs21e I saw your comment on #829 (https://github.com/patternfly/patternfly-design/issues/829#issuecomment-607315823) and agree that it is not really a dropdown that you want. Some kind of tree menu makes the most sense.  I do believe that we have this on our roadmap, so glad to collaborate on that design.

I'm actually going to close this to prevent confusion. We had abandoned this request in favor of addressing the Topology use case with the context menu as described here: https://github.com/patternfly/patternfly-design/issues/774
